client/src/partials/journal/modals/journalPosting.modal.js

```diff
@@ -0,0 +1,124 @@
+angular.module('bhima.controllers')
+  .controller('JournalPosterModalController', JournalPosterModalController);
+
+JournalPosterModalController.$inject = [ 
+  '$uibModalInstance', 'SessionService', 'JournalPostingModalService',
+  'GridGroupingService', 'Records', 'GridColumnService', 'NotifyService',
+  '$location'
+];
+
+/**
+ * @module journal/modals/JournalPoster.modal
+ *
+ * @description
+ * This controller provides a tool to do trial balance
+ */
+function JournalPosterModalController(ModalInstance, Session, journalPostingModalService, Grouping,  records, Columns, Notify, $location) {
+  var vm = this;
+  var columns = [
+    { field : 'trans_id', displayName : 'TABLE.COLUMNS.TRANSACTION', headerCellFilter: 'translate', enableCellEdit: false, allowCellFocus: false},
+    { field : 'account_number', displayName : 'TABLE.COLUMNS.ACCOUNT', headerCellFilter: 'translate'},
+    { field : 'balance_before', displayName : 'TABLE.COLUMNS.BEFORE', headerCellFilter : 'translate', enableCellEdit : false, allowCellFocus : false, visible : false},
+    { field : 'debit_equiv', displayName : 'TABLE.COLUMNS.DEBIT', headerCellFilter: 'translate', cellTemplate : '/partials/templates/grid/debit_equiv.cell.html' },
+    { field : 'credit_equiv', displayName : 'TABLE.COLUMNS.CREDIT', headerCellFilter: 'translate', cellTemplate : '/partials/templates/grid/credit_equiv.cell.html'},
+    { field : 'balance_final', displayName : 'TABLE.COLUMNS.AFTER', headerCellFilter : 'translate', visible : false}
+  ];
+
+  vm.enterprise = Session.enterprise;
+  vm.errors = null;
+  vm.cssClass = null;
+  vm.feedBack = null;
```

Comment:
It is just my way of doing, I prefer to initialize every property or variable by a default value instead of having ```undefined```. But no problem I can delete them.